src/actions/log.ts

```diff
@@ -0,0 +1,19 @@
+import {PlainReduxAction, ReduxAction} from './redux-action';
+
+export type LogAction = LogErrorsAdd | LogWarningsAdd | LogWarningsRemove | LogErrorsRemove;
+
+export const LOG_ERRORS_ADD = 'LOG_ERRORS_ADD';
+export type LogErrorsAdd = ReduxAction<typeof LOG_ERRORS_ADD, {
+  errors: string[]
+}>;
+
+export const LOG_WARNINGS_ADD = 'LOG_WARNINGS_ADD';
+export type LogWarningsAdd = ReduxAction<typeof LOG_WARNINGS_ADD, {
+  warnings: string[]
+}>;
+
+export const LOG_WARNINGS_REMOVE = 'LOG_WARNINGS_REMOVE';
```

Comment:
Right now warnings and errors are displayed in two separate pane (one in red and one in yellow). Do you think we should show them in one pane instead?